### DISCUSSION TITLE
Unable to store excel spreadsheet with graph/chart as PDF file.

### DISCUSSION THREAD

**Discussion by boesen** [PHPOffice/PhpSpreadsheet#4168] [Q&A]
Hi, 

I'm using PHP and PhpSpreadsheet to create statistic files, now I need to add some graph/chart to the statistic. 

It works fine creating the xlsx file with data and charts - but when I try to store the spreadsheet as a PDF file the chart is missing. 

I have subsequently read that PhpSpreadsheet don't support storing chart to PDF but it was in some old articles. Is that still a issue?

If so, does anyone have a solution to embed chart into the xlsx files so it can be stored as a PDF file. 

Thanks in advance.

/Torben

**Comment by oleibman** [comment]
PhpSpreadsheet can write spreadsheets with charts to PDF. Before writing, you need to identify a chart renderer, e.g.
```php
\PhpOffice\PhpSpreadsheet\Settings::setChartRenderer(
    PhpOffice\PhpSpreadsheet\Chart\Renderer\MtJpGraphRenderer::class
);
```
And you must tell the writer to include charts:
```php
$writer->setIncludeCharts(true);
```

**Reply by boesen** [reply]
Okay, tried to add the JpGraphRenderer - which added the chart to the PDF files, but the size of the chart was different compare to the size in the spreadsheet - it look like it was added to one cell only.

In the php scrip the chart size is set to 
    $chart->setTopLeftPosition('A40');
    $chart->setBottomRightPosition('G60');

but it look like it is only inserted into A40

Is there any size settings for the rendered chart object?

/Torben

**Reply by oleibman** [reply]
This is a known problem. See #4129 for some possible mitigations.

**Comment by oleibman** [comment]
If you can test against PR #4327, adding calls to the new methods introduced there, please do so.

**Comment by oleibman** [comment]
PR #4327 is now merged, providing a solution to this problem.
